bsp/src/org/jetbrains/bsp/BspServerWidgetProvider.scala

```diff
@@ -89,11 +98,51 @@ class BspServerWidgetProvider extends StatusBarWidgetProvider {
       }
     }
 
+    class RestartServer extends AnAction(BspBundle.message("bsp.widget.stop.all.bsp.restart"), BspBundle.message("bsp.widget.stop.all.bsp.restart"), AllIcons.Actions.Exit) with DumbAware {
+
+      override def update(e: AnActionEvent): Unit = {
+
+      }
+
+      override def actionPerformed(e: AnActionEvent): Unit = try {
+        BspCommunicationService.getInstance.closeAll
+        ApplicationManager.getApplication.executeOnPooledThread(
+          new Runnable {
+            override def run(): Unit = {
+              val commands = bloopExitCommands(project).flatten
+              if (commands.isEmpty) {
+                logger.warn(s"BSP server exit command not set in '${argvExit}' field")
```

Comment:
`argvExit` isn't defined in BSP specification and not required, shouldn't really be a warning.